### DISCUSSION TITLE
Missing Washer Program

### DISCUSSION THREAD

**Discussion by hmichels** [ekutner/home-connect-hass#167] [Community Support]
Hello!
Thanks for this great integration, I am just starting to control my two devices.

I am using the latest public release with a washing machine that offers "LaundryCare.Washer.Program.Cotton" as well as "LaundryCare.Washer.Program.Cotton.Colour" washing program.

If I set "Cotton Color" on the device, the entities "sensor.siemens_wm16xe91_68a40e5471bb_active_program", "sensor.siemens_wm16xe91_68a40e5471bb_selected_program" and "select.siemens_wm16xe91_68a40e5471bb_selected_program" show "unknown".

Here I could enable the correct display of "sensor.siemens_wm16xe91_68a40e5471bb_selected_program" by adding entry "LaundryCare.Washer.Program.Cotton.Colour" with translation to the files "sensors_en.yaml", "sensors_de.yaml", "select_en.yaml" and "select_de.yaml". But this not work for "select.siemens_wm16xe91_68a40e5471bb_selected_program". The program "Cotton Color" cannot be selected and therefore not started.

With the official "Home Connect" APP I'm able to select and to start a program for "Cotton Color". 
In this case "sensor.siemens_wm16xe91_68a40e5471bb_selected_program" show the correct value "Cotton Color", but "sensor.siemens_wm16xe91_68a40e5471bb_active_program" and "select.siemens_wm16xe91_68a40e5471bb_selected_program" show still "unknown".

Can I be helped here? What can I try?

It would also be nice if someone can show an example (YAML) of automating the appliances using, for example, the "program", the "temperature" and the "spin speed" as options to start the washing machine.

Thanks a lot!
Greetings, Holger

**Comment by ekutner** [comment]
I don't know why, but there are programs that are not supported by the Home Connect API, so you will not be able to start them through the integration. The missing translation is unrelated, if the program was supported you would have seen it in the select box like you did in the sensor.

**Reply by hmichels** [reply]
Thanks for your replay!
But when the official "Home Connect" APP is able to select and to start a program for "Cotton Color", the Home Connect API support this program, or am I wrong about that?

**Reply by ekutner** [reply]
The Home Connect mobile app is using a proprietary API which is not available to the public and has additional functionality so the fact something is supported there doesn't mean anything for their public API.

**Reply by hmichels** [reply]
Okay, didn't think the API was so "restricted" 😥.

What about an example (YAML) of an automation, how to use the "program", the "temperature" and the "spin speed" as options to start the washing machine?
I still use the service for program starting without optional "data".

Many thanks!
Regards, Holger
